src/lib/publicsuffixlist/publicsuffixlist.js

```diff
@@ -540,7 +540,8 @@ const enableWASM = (( ) => {
         if ( uint8s[0] !== 1 ) { return false; }
 
         try {
-            const module = await wasmModuleFetcher(`${path}publicsuffixlist`);
+            const { default: bytes } = await import('./wasm/publicsuffixlist.wasm.js');
```

Comment:
I want to wait next dev cycle for this. This adds a step in how the wasm code is generated (see local README file), and this could cause an issue when I submit uBO to the AMO store -- the reviewers will want to validate that the content of the array are as expected the output of the `wat2wasm` utility.

Also, unlike when using `compileInstance`, the current way opens the door of the extension modifying the wasm code before compiling it, essentially the equivalent of having inline code in the extension, which is what I wanted to avoid on principle by using `compileInstance` so that code reviewers can trust the wasm code really what the `.wasm` files contain.

Reply:
Makes sense.

Regarding the upstream [Node.js package](https://github.com/gorhill/publicsuffixlist.js), let me see then if I can come up with something that is also compatible with these requirements, because I like the design that client code does not have to pass its own `wasmModuleFetcher` implementation, which is how it is now over there.

The main limitation over there is that `fetch()` is not available on Node.js.

I'll close this for now and try to find a way to address it over there.

Reply:
@gorhill is it fair to say that the implementations of `enableWASM()` and `disableWASM()` over here are newer than the ones in the upstream [Node.js package](https://github.com/gorhill/publicsuffixlist.js) and therefore it's OK to simply copy these over to that package as a start?

Reply:
Yes, the changes here are more recent than the official repo.